### DISCUSSION TITLE
Quick info on ragdolls

### DISCUSSION THREAD

**Discussion by ThomasCartier** [jrouwe/JoltPhysics#802] [Q&A] [answered]
Hi,
Just finished setting the skeletons, their animations.
Now, to add the limbs of the body, I plan to use a ragdoll. 
I don't see any alternative, but I may be wrong.

The question is : why should I set the ragdoll's twist constraints, if I apply a kinematic animation? 
(Here is the creation [sample,](https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/Rig/CreateRigTest.cpp#L185) but the kinematic loads the animation and human with these settings set as well).

Shouldn't it override any settings set there ? 
I mean, If I have a humanoid with a walking animation, the limbs are forced to their final position/rotation by the SetPose call each frame. Irrespective of any constraint (?)

```
mPose.SetSkeleton(mRagdollSettings->GetSkeleton());

// Position ragdoll
mAnimation->Sample(0.0f, mPose);
mPose.CalculateJointMatrices();
mRagdoll->SetPose(mPose);
```
May these settings be left unset? Is there any incidence with the physics system by any mean?
Do I miss something ?

**Comment by jrouwe** [comment, marked as answer]
If you plan to only use your ragdoll as a kinematic object only, then there's no reason to create any constraints (constraints only do something when one of the connected bodies is dynamic).

Note that you should only use SetPose to put the ragdoll in its initial pose, after that you should use DriveToPoseUsingKinematics to drive your kinematic ragdoll (see the KinematicRigTest sample).

**Reply by ThomasCartier** [reply]
Nice, thank a lot !
